### PR TITLE
feat(rule): accessibility rule for alt text

### PR DESCRIPTION
**Pull request by mohammedzamakhan**

### CODE REVIEW

File: src/templateAccessibilityAltTextRule.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import { SourceFile } from 'typescript/lib/typescript';
+import { NgWalker } from './angular/ngWalker';
+import { BasicTemplateAstVisitor } from './angular/templates/basicTemplateAstVisitor';
+
+export class Rule extends Rules.AbstractRule {
+  static readonly metadata: IRuleMetadata = {
+    description: 'Ensures that all elements that require alternate text i.e. img, area, input[type="image"] and object.',
+    options: null,
+    optionsDescription: 'Not configurable.',
+    rationale: 'Alternate text give meaningful information to relay back to the end user',
```
Comment (mgechev) [inline src/templateAccessibilityAltTextRule.ts]: Alternate text lets screen readers provide more information to end users?

File: src/templateAccessibilityAltTextRule.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import { ElementAst, AttrAst, BoundElementPropertyAst, TextAst } from '@angular/compiler';
+import { sprintf } from 'sprintf-js';
+import { IRuleMetadata, RuleFailure, Rules } from 'tslint/lib';
+import { SourceFile } from 'typescript/lib/typescript';
+import { NgWalker } from './angular/ngWalker';
+import { BasicTemplateAstVisitor } from './angular/templates/basicTemplateAstVisitor';
+
+export class Rule extends Rules.AbstractRule {
+  static readonly metadata: IRuleMetadata = {
+    description: 'Ensures that all elements that require alternate text i.e. img, area, input[type="image"] and object.',
```
Comment (mgechev) [inline src/templateAccessibilityAltTextRule.ts]: Enforces alternate text for elements which require alt attribute

File: src/templateAccessibilityAltTextRule.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    if (isValid) {
+      return;
+    }
+    const {
+      sourceSpan: {
+        end: { offset: endOffset },
+        start: { offset: startOffset }
+      }
+    } = element;
+    console.log(element.name);
```
Comment (mgechev) [inline src/templateAccessibilityAltTextRule.ts]: You can drop the log

File: src/templateAccessibilityAltTextRule.ts (hunk 2 above)
Comment (mohammedzamakhan) [reply src/templateAccessibilityAltTextRule.ts]: Its not just alt attribute @mgechev, check the test cases

**Pull request merged**